sdk/mobile-center-updates/src/test/java/com/microsoft/azure/mobile/updates/UpdatesBeforeDownloadTests.java

```diff
@@ -0,0 +1,526 @@
+package com.microsoft.azure.mobile.updates;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.pm.PackageManager;
+
+import com.microsoft.azure.mobile.channel.Channel;
+import com.microsoft.azure.mobile.http.HttpClient;
+import com.microsoft.azure.mobile.http.HttpClientNetworkStateHandler;
+import com.microsoft.azure.mobile.http.ServiceCall;
+import com.microsoft.azure.mobile.http.ServiceCallback;
+import com.microsoft.azure.mobile.utils.AsyncTaskUtils;
+
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InOrder;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+
+import java.util.HashMap;
+import java.util.concurrent.Semaphore;
+
+import static com.microsoft.azure.mobile.updates.UpdateConstants.PREFERENCE_KEY_DOWNLOAD_URI;
+import static com.microsoft.azure.mobile.updates.UpdateConstants.PREFERENCE_KEY_UPDATE_TOKEN;
+import static com.microsoft.azure.mobile.utils.storage.StorageHelper.PreferencesStorage;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyMapOf;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static org.mockito.internal.verification.VerificationModeFactory.times;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.verifyStatic;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+
+public class UpdatesBeforeDownloadTests extends AbstractUpdatesTest {
+
+    @Test
+    public void failsToCompareVersion() throws Exception {
+
+        /* Mock we already have token. */
```

Comment:
No you can't override setUp, it leads to suprising results with the annotations.